src/main/java/gregtech/api/terminal/os/TerminalOSWidget.java

```diff
@@ -163,7 +163,7 @@ public void openApplication(AbstractApplication application, boolean isClient) {
             closeApplication(focusApp, isClient);
         }
         for (AbstractApplication app : openedApps) {
-            if (app.getClass() == application.getClass()) {
+            if (app.getClass() == application.getClass() && application.canLaunchConcurrently(app)) {
```

Comment:
change `app.getClass() == application.getClass()` to `app.getRegistryName().equals(application.getRegistryName())`